File: src/main/java/aurocosh/divinefavor/common/potions/base/potion/ModPotionToggle.java
package aurocosh.divinefavor.common.potions.base.potion;

import net.minecraft.client.Minecraft;
import net.minecraft.client.resources.I18n;
import net.minecraft.potion.PotionEffect;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

public abstract class ModPotionToggle extends ModPotion {
    public ModPotionToggle(String name, boolean beneficial, int potionColor) {
        super(name, beneficial, potionColor);
    }

    @Override
    public boolean shouldRenderInvText(PotionEffect effect) {
        return false;
    }

    @SideOnly(Side.CLIENT)
    @Override
    public void renderCustomInvText(int x, int y, PotionEffect effect, Minecraft mc) {
        String potionName = I18n.format(getName());
        mc.fontRenderer.drawStringWithShadow(potionName, (float) (x + 10 + 18), (float) (y + 6), 16777215);
        mc.fontRenderer.drawStringWithShadow("Permament", (float) (x + 10 + 18), (float) (y + 6 + 10), 8355711);
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/potions/base/potion/ModPotionToggle.kt
package aurocosh.divinefavor.common.potions.base.potion

import net.minecraft.client.Minecraft
import net.minecraft.client.resources.I18n
import net.minecraft.potion.PotionEffect
import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly

abstract class ModPotionToggle(name: String, potionColor: Int) : ModPotion(name, potionColor) {

    override fun shouldRenderInvText(effect: PotionEffect?): Boolean {
        return false
    }

    @SideOnly(Side.CLIENT)
    override fun renderCustomInvText(x: Int, y: Int, effect: PotionEffect, mc: Minecraft) {
        val potionName = I18n.format(name)
        mc.fontRenderer.drawStringWithShadow(potionName, (x + 10 + 18).toFloat(), (y + 6).toFloat(), 16777215)
        mc.fontRenderer.drawStringWithShadow("Permanent", (x + 10 + 18).toFloat(), (y + 6 + 10).toFloat(), 8355711)
    }
}
